source/core/Ky.ts

```diff
@@ -149,16 +151,22 @@ export class Ky {
 			throw new TypeError('`input` must be a string, URL, or Request');
 		}
 
-		if (this._options.prefixUrl && typeof this._input === 'string') {
-			if (this._input.startsWith('/')) {
-				throw new Error('`input` must not begin with a slash when using `prefixUrl`');
-			}
+		if (typeof this._input === 'string') {
+			if (this._options.startPath) {
+				if (!this._options.startPath.endsWith('/')) {
+					this._options.startPath += '/';
+				}
 
-			if (!this._options.prefixUrl.endsWith('/')) {
-				this._options.prefixUrl += '/';
+				if (this._input.startsWith('/')) {
+					this._input = this._input.slice(1);
+				}
+
+				this._input = this._options.startPath + this._input;
```

Comment:
I think if `_input` starts with a protocol (or is a valid URL?) then we should ignore `startPath` in order to avoid an URL like `http://foo.com/http://bar.com`

Reply:
That kind of "double URL" construct is useful for reverse proxies. It should be supported.

Also, you would be surprised how hard URL parsing is. Especially for a library like Ky that supports non-browsers.